### DISCUSSION TITLE
Run MCP Tools with Any OpenAI-Compatible LLM

### DISCUSSION THREAD

**Discussion by bartolli** [modelcontextprotocol/.github#60] [Show and tell]
### Pre-submission Checklist

- [X] This post follows the MCP community guidelines

### What would you like to share?

Hi guys,

I built an implementation that brings MCP's tooling system to OpenAI's function calling interface. The bridge enables using MCP-compliant tools with OpenAI and other OpenAPI-compatible models, extending MCP's reach beyond Claude Desktop.

While it's primarily designed for use with the OpenAI API, you can also use it with Ollama running locally, LM Studio, or any endpoint that implements the OpenAI API spec.

It's a contribution toward broader MCP adoption and interoperability in the LLM ecosystem.

### Relevant Links

Check out [the demo and code here](https://github.com/bartolli/mcp-llm-bridge) and let me know what you think.

**Comment by amira-essawy** [comment]
@bartolli I don't have OpenAI key, can I run it using Ollama without key?
Also, is it supported to use gemini?

**Comment by gl2007** [comment]
@bartolli this is very interesting and was looking for something along these lines. Is it possible to get this under mcp repository?
